server.py

```diff
@@ -43,55 +43,69 @@ def first_of_milestone(feature_list, milestone, start=0):
       return i
+
   return -1
 
+def first_of_milestone_v2(feature_list, milestone, start=0):
+  for i in xrange(start, len(feature_list)):
+    f = feature_list[i]
+    desktop_milestone = f['browsers']['chrome'].get('desktop', None)
+    android_milestone = f['browsers']['chrome'].get('android', None)
+    status = f['browsers']['chrome']['status'].get('text', None)
 
-class MainHandler(http2push.PushHandler, common.ContentHandler, common.JSONHandler):
+    if (str(desktop_milestone) == str(milestone) or status == str(milestone)):
+      return i
+    elif (desktop_milestone == None and str(android_milestone) == str(milestone)):
+      return i
 
-  def __get_omaha_data(self):
-    omaha_data = memcache.get('omaha_data')
-    if omaha_data is None:
-      result = urlfetch.fetch('https://omahaproxy.appspot.com/all.json')
-      if result.status_code == 200:
-        omaha_data = json.loads(result.content)
-        memcache.set('omaha_data', omaha_data, time=86400) # cache for 24hrs.
-
-    return omaha_data
-
-  def __annotate_first_of_milestones(self, feature_list):
-    try:
-      omaha_data = self.__get_omaha_data()
-
-      win_versions = omaha_data[0]['versions']
-      for v in win_versions:
-        s = v.get('version') or v.get('prev_version')
-        LATEST_VERSION = int(s.split('.')[0])
-        break
-
-      # TODO(ericbidelman) - memcache this calculation as part of models.py
-      milestones = range(1, LATEST_VERSION + 1)
-      milestones.reverse()
-      versions = [
-        models.IMPLEMENTATION_STATUS[models.NO_ACTIVE_DEV],
-        models.IMPLEMENTATION_STATUS[models.PROPOSED],
-        models.IMPLEMENTATION_STATUS[models.IN_DEVELOPMENT],
-        ]
-      versions.extend(milestones)
-      versions.append(models.IMPLEMENTATION_STATUS[models.NO_LONGER_PURSUING])
-
-      last_good_idx = 0
-      for i, version in enumerate(versions):
-        idx = first_of_milestone(feature_list, version, start=last_good_idx)
-        if idx != -1:
-          feature_list[idx]['first_of_milestone'] = True
-          last_good_idx = idx
-    except Exception as e:
-      logging.error(e)
-
-  def __get_feature_list(self):
-    feature_list = models.Feature.get_chronological() # Memcached
-    self.__annotate_first_of_milestones(feature_list)
-    return feature_list
+  return -1
+
+def get_omaha_data():
+  omaha_data = memcache.get('omaha_data')
+  if omaha_data is None:
+    result = urlfetch.fetch('https://omahaproxy.appspot.com/all.json')
+    if result.status_code == 200:
+      omaha_data = json.loads(result.content)
+      memcache.set('omaha_data', omaha_data, time=86400) # cache for 24hrs.
+
+  return omaha_data
+
+def annotate_first_of_milestones(feature_list, version=None):
+  try:
+    omaha_data = get_omaha_data()
+
+    win_versions = omaha_data[0]['versions']
+    for v in win_versions:
+      s = v.get('version') or v.get('prev_version')
+      LATEST_VERSION = int(s.split('.')[0])
+      break
```

Comment:
`win_versions[0]` is an array. 

Looks like they've changed the data format a bit since I last looked at this. I've gone ahead and simplified how the latest version is found by using a list comprehension to find the canary channels in the array, then using the first's `version`.